network/netplan/netplan_test.go

```diff
@@ -603,3 +603,40 @@ func (s *NetplanSuite) TestWriteCantGenerateName(c *gc.C) {
 	_, err = np.Write("")
 	c.Check(err, gc.ErrorMatches, "Can't generate a filename for netplan YAML")
 }
+
+func (s *NetplanSuite) TestProperReadingOrder(c *gc.C) {
+	var header = `
+network:
+  version: 2
+  renderer: NetworkManager
+  ethernets:
+`[1:]
+	var template = `
+    id%d:
+      match: {}
+      set-name: foo.%d.%d
+`[1:]
+	tempDir := c.MkDir()
+
+	for _, n := range rand.Perm(100) {
+		content := header
+		for i := 0; i < (100 - n); i++ {
+			content += fmt.Sprintf(template, i, i, n)
+		}
+		ioutil.WriteFile(path.Join(tempDir, fmt.Sprintf("%0.2d-test.yaml", n)), []byte(content), 0644)
```

Comment:
out of curiosity, do we have any idea how netplan would sort
foo-1.yaml
vs
foo-2.yaml
vs
foo-10.yaml
?
I'm guessing it just does lexical sort, and not natural sort, but I'd be curious to know for sure.

Also, did you find that the ReadDir always gave back sorted order, or was this test failing without the sortableFileInfo step?